lib/std/array_list.zig

```diff
@@ -335,6 +362,33 @@ pub fn ArrayListAlignedUnmanaged(comptime T: type, comptime alignment: ?u29) typ
             mem.copy(T, self.items[i .. i + items.len], items);
         }
 
+        /// Replace range of elements `list[start..start+len]` with `items`
+        /// grows list if `len < items.len`. may allocate
+        /// shrinks list if `len > items.len`
+        pub fn replaceRange(self: *Self, allocator: *Allocator, start: usize, len: usize, items: SliceConst) !void {
```

Comment:
You don't have to copy the impl for ArrayListUnmanaged. This pattern works fine:

```zig
var managed = self.toManaged(allocator);
try managed.replaceRange(start, len, items);
return managed.toUnmanaged();
```

The reason all the functions don't do that is that most of them are less than 3 lines anyway.

Reply:
makes sense